src/v/kafka/server/connection_context.h

```diff
@@ -106,7 +106,9 @@ class connection_context final
 
     protocol& server() { return _proto; }
     const ss::sstring& listener() const { return _rs.conn->name(); }
-    security::sasl_server& sasl() { return _sasl; }
+    security::sasl_server* sasl() {
+        return _sasl.has_value() ? &_sasl.value() : nullptr;
+    }
```

Comment:
I'm not sure what you're suggesting. Would returning a `std::optional<std::reference_wrapper<sasl_server>>` be less objectionable?

Reply:
I'm suggesting to use the reference returned by `std::optional::value` instead of a pointer to the value. In that case this function wouldn't be required at all.

Reply:
OK, but how do I write this? https://github.com/redpanda-data/redpanda/pull/5906/commits/7878fa0e354c130636e3ac66be4c9b635963c676#diff-31a264c0336551049114643921a848dd776a048d98fab6176a58c50f96b398f8R52-R56
```c++
          if (unlikely(
                sasl()
                && sasl()->state()
                     == security::sasl_server::sasl_state::authenticate
                && sasl()->handshake_v0())) 
```

Reply:
Unless I'm entirely missing something.

```
std::optional<security::sasl_server>& sasl() { return _sasl; }

if (unlikely(ctx.sasl().has_value() && !ctx.sasl().value().complete())) {
```

Reply:
Done